### ISSUE TITLE
[Bug] [NPU] Ascend A3 inference of MiniMax H3 has precision issues

### ISSUE THREAD

**Issue by OrangePure** [sgl-project/sglang#39386]
### Checklist

- [x] I searched related issues but found no solution.
- [x] The bug persists in the latest version.
- [x] Issues without environment info and a minimal reproducible demo are hard to resolve and may receive no feedback.
- [x] If this is not a bug report but a general question, please start a discussion at https://github.com/sgl-project/sglang/discussions. Otherwise, it will be closed.
- [x] Please use English. Otherwise, it will be closed.

### Describe the bug

I referenced the cookbook, using A3 4-NPU and 8-NPU configurations to run inference on MiniMax H3 official weights. T2VA, FL2VA, and Ref2VA all produce gray-noise screens and noise. Is there a code version + configuration that has been verified on A3?

Bug description

   Every generation completes without error and produces a well-formed MP4 (correct resolution/duration/streams), but the video content is uniform brown noise on every frame and the audio is noise with an abnormal DC offset
   (~0.09–0.10). Latency numbers look normal (e.g. 768p 5s ≈ 46s on 8 NPUs, matching the cookbook benchmark).

   Reproduced across all of these variations — output is always noise:

   • t2va and ref2va tasks
   • 768p and 1080p, 4s/5s/10s durations, 20/50 steps
   • laser_attn and fa attention backends
   • 8-NPU (TP2+SP4), 16-NPU (TP2+Ulysses4+Ring2), resident and layerwise-offload residency
   • Cache-DiT enabled and disabled (SGLANG_CACHE_DIT_ENABLED unset)

   Weights ruled out: we byte-verified the local weights two ways — (1) all small files + an official-layout copy match the ModelScope repo SHA256s; (2) the transformer (66GB, both FL2VA and Ref2VA), text_encoder (902 tensors),
   video_vae (560) and audio_vae (743) are byte-identical to a local ComfyUI model copy that generates correct 1080p videos on this same machine. So the noise originates in the sglang runtime path, not the weights.

   Notable context:

   • Machine is Atlas 800I A3 (16 dies visible), while issue #33357 marks A3 support as WIP and its reference config uses use_fsdp_inference=True, sp_degree=8, ulysses_degree=8, vae_cpu_offload=True (via DiffGenerator), which differs
     from the cookbook's TP2+SP4 serve config.
   • A known-problem area: the VAE klvae.py tiled encode/decode under sequence parallelism (an SP collective participation fix exists in one of our local copies).
   • Two local patches applied (for unrelated issues): resolved_plan.py — made MINIMAX_H3_MAX_PIXELS env-configurable; qwen3vl.py:872 — position_ids.masked_fill_(mask==0, 1) → torch.where(...) because the inplace scalar fill crashes
     with aclnnInplaceFillScalar on torch_npu 2.10 + CANN 9.0.1.

### Reproduction
Command/script run and model used
   Model: MiniMax/MiniMax-H3 (ModelScope), variant FL2VA, loaded from local path /data_work/models/MiniMax-H3.
   Server launch (8× Ascend910_9382, following the cookbook):
   ```bash
     export SGLANG_MINIMAX_H3_MAX_PIXELS=2088960   # local patch to allow 1080p
     export HCCL_BUFFSIZE=256
     sglang serve \
       --model-path /data_work/models/MiniMax-H3 \
       --model-type diffusion \
       --model-variant fl2va \
       --dit-cpu-offload false \
       --num-gpus 8 \
       --tp-size 2 \
       --sp-degree 4 \
       --attention-backend laser_attn \
       --port 30088 \
       --component-residency text_encoder=layerwise-offload
   ```

   Client request:
   ```bash
     curl -X POST http://127.0.0.1:30088/v1/videos -H "Content-Type: application/json" -d '{
       "model": "/data_work/models/MiniMax-H3",
       "prompt": "At night, while their owner sleeps in a bedroom, three cats march in loudly playing tiny brass instruments, then abruptly file out.",
       "seconds": 5,
       "task": "t2va",
       "conditions": [],
       "target": {"short_edge": 768, "aspect_ratio": "16:9", "duration_seconds": 5.0},
       "num_outputs_per_prompt": 1,
       "num_inference_steps": 50,
       "flow_shift": 12.0,
       "audio_flow_shift": 3.0,
       "seed": 1101
     }'
   ```

### Environment
Run python3 -m sglang.check_env and paste output here
```markdown
     Python: 3.11.0rc1 (main, Jun 29 2026, 00:05:30) [GCC 11.4.0]
     NPU available: True
     NPU 0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15: Ascend910_9382
     CANN_HOME: /usr/local/Ascend/cann-9.0.1
     CANN: Not Available
     BiSheng: 2026-04-25T15:46:25+08:00 clang version 15.0.5 (clang-5c68a1cb1231 flang-5c68a1cb1231)
     Ascend Driver Version: 26.1.1
     PyTorch: 2.10.0+cpu
     sglang: 0.0.0   (editable install from source tree, ~v0.5.19)
     sglang-kernel: Module Not Found
     flashinfer_python: Module Not Found
     flashinfer_cubin: Module Not Found
     flashinfer_jit_cache: Module Not Found
     triton: 3.5.0
     transformers: 5.12.1
     torchao: 0.9.0
     numpy: 1.26.4
     aiohttp: 3.14.3
     fastapi: 0.141.1
     huggingface_hub: 1.31.0
     interegular: 0.3.3
     modelscope: 1.40.0
     orjson: 3.12.0
     outlines: 0.1.11
     packaging: 26.3
     psutil: 6.0.0
     pydantic: 2.13.5
     python-multipart: 0.0.32
     pyzmq: 27.2.0
     uvicorn: 0.52.4
     uvloop: 0.22.1
     vllm: Module Not Found
     xgrammar: 0.2.1
     openai: 2.6.1
     tiktoken: 0.14.0
     anthropic: 1.5.0
     litellm: Module Not Found
     torchcodec: Module Not Found
     torch_npu: 2.10.0
     sgl-kernel-npu: 2026.9.0
     deep_ep: 1.0.0+d9261669.cann.9.0.0.b250
     Ascend Topology: (16× Ascend910_9382 — full matrix below)
     ulimit soft: 1024
   ```

**Comment by OrangePure** [comment]
@iforgetmyname
